sympy/solvers/bivariate.py

```diff
@@ -208,8 +214,16 @@ def _solve_lambert(f, symbol, gens):
     if lhs.is_Mul:
         lhs = expand_log(log(lhs))
         rhs = log(rhs)
+        if lhs.is_Add and lhs.has(t):
+            llhs1, llhs2 = map(lambda i: lhs.xreplace({t: i}), (symbol, -symbol))
+            sol1, sol2 = map(lambda i: _solve_lambert(i - rhs, symbol, gens), (llhs1, llhs2))
+            return list((set(sol1 + sol2)))
```

Comment:
Just do simple concatenation: `return sol1 + sol2`.

Reply:
I applied set because same solution were being returned twice in a list in some tests.

Reply:
Interesting! Any specific reason? Maybe the same equation is being solved multiple times. Probably this needs to be fixed.

Reply:
Only one test of `test_ode` gets failed. Here is the test and explanation:-
```python
>>>eq5 = 2*x**2*f(x) + f(x)**3 + (x*f(x)**2 - 2*x**3)*f(x).diff(x)
>>>dsolve(eq5, hint='1st_homogeneous_coeff_best')
```
In this test case the equation that enters `_solve_lambert`  is 
```python
# lhs = _X0**2*(-2*C1 + 2*log(_X0) + log(x/_X0)) and rhs = -x**2 where
# symbol = _X0
>>>symbol.assumptions0
{'commutative': True}
>>>symbol.is_positive
None
```
So when in this equation _X0 gets replaced by t (becomes `_t**2*(-2*C1 + 2*log(_X0) + log(x/_X0))`) it enters code of `lhs.is_Mul`
where after `lhs = expand_log(log(lhs))` happens then lhs becomes
`log(_t**2*(-2*C1 + 2*log(_X0) + log(x/_X0)))`
Clearly `_t` doesn't have assumptions to be positive so it doesn't get separated out but the equation gets solved twice with -symbol and +symbol where both equation comes out to be same.
I think its difficult to separate out those t's which can't be separated out or to determine. One thing we can do here that from assumptions checking out whether symbol.is_positive or not. But I think that will be too specific. But if you advise then I will try to do something to avoid same equations. Although it happens only when dsolve calls because in solve equations enter `_solve_lambert` with symbol assumptions positive. In `solve it doesn't have same equation solved twice.